ecl/hqlcpp/hqlttcpp.cpp

```diff
@@ -3528,6 +3528,68 @@ IHqlExpression * ThorHqlTransformer::normalizeMergeAggregate(IHqlExpression * ex
 }
 
 
+static bool reorderAggregateFields(HqlExprArray & aggregateFields, HqlExprArray & aggregateAssigns)
+{
+    bool needToReorder = false;
+    bool variableOffset = false;
+    bool dynamicOffset = false;
+    ForEachItemIn(i, aggregateFields)
+    {
+        IHqlExpression & cur = aggregateFields.item(i);
+        IHqlExpression * value = aggregateAssigns.item(i).queryChild(1);
+        if (value->isGroupAggregateFunction())
+        {
+            if (dynamicOffset)
+                needToReorder = true;
+        }
+
+        if (isUnknownSize(&cur))
+        {
+            if (value->isGroupAggregateFunction())
+                dynamicOffset = true;
+            else
+                variableOffset = true;
+        }
+    }
+
+    if (!needToReorder)
+        return false;
+
+    HqlExprArray newFields;
+    HqlExprArray newAssigns;
+    for (unsigned pass = 0; pass < 3; pass++)
+    {
+        //Fixed size fields first, then non aggregates, then aggregates
+        ForEachItemIn(i1, aggregateFields)
+        {
+            IHqlExpression & cur = aggregateFields.item(i1);
+            IHqlExpression & assign = aggregateAssigns.item(i1);
+            bool copy = false;
+            if (!isUnknownSize(&cur))
```

Comment:
Ok, I'm not sure of the logic regarding exclusivity here either. Is it really intended for (!isUnknownSize(&cur) && assign.queryChild(1)->isGroupAggregateFunction()) to be ignored?